From 4664225c3d7d11c9ab6042a4e41e476217174b63 Mon Sep 17 00:00:00 2001
From: Yufei Gu <yufei@apache.org>
Date: Fri, 16 Mar 2018 16:25:33 -0700
Subject: [PATCH 2781/2848] YARN-8039. Clean up log dir configuration in
 TestLinuxContainerExecutorWithMocks.testStartLocalizer.
 Contributed by Miklos Szegedi.

(cherry picked from commit 49c747ab187d0650143205ba57ca19607ec4c6bd)
(cherry picked from commit 204674f4146b2869618c1c3ad58a140febc4df28)

Conflicts:
	hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
Just a simple merge to a different location in a unit test.

Change-Id: Iaebb93fcef1cd3c913ddb1308f1683adfc31c795
(cherry picked from commit 95e066222be2f53d743e0b2eb573b2ecb1cd13f2)
---
 .../TestLinuxContainerExecutorWithMocks.java       |   16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
index fcb1c8a..557d67f 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
@@ -84,7 +84,7 @@ private void deleteMockParamFile() {
     reader.close();
     return ret;
   }
-  
+
   @Before
   public void setup() {
     assumeTrue(!Path.WINDOWS);
@@ -95,6 +95,7 @@ public void setup() {
     String executorPath = f.getAbsolutePath();
     Configuration conf = new Configuration();
     conf.set(YarnConfiguration.NM_LINUX_CONTAINER_EXECUTOR_PATH, executorPath);
+    conf.set(YarnConfiguration.NM_LOG_DIRS, "src/test/resources");
     mockExec = new LinuxContainerExecutor();
     dirsHandler = new LocalDirsHandlerService();
     dirsHandler.init(conf);
@@ -105,7 +106,7 @@ public void setup() {
   public void tearDown() {
     deleteMockParamFile();
   }
-  
+
   @Test
   public void testContainerLaunch() throws IOException {
     String appSubmitter = "nobody";
@@ -117,7 +118,7 @@ public void testContainerLaunch() throws IOException {
     ContainerId cId = mock(ContainerId.class);
     ContainerLaunchContext context = mock(ContainerLaunchContext.class);
     HashMap<String, String> env = new HashMap<String,String>();
-    
+
     when(container.getContainerId()).thenReturn(cId);
     when(container.getLaunchContext()).thenReturn(context);
     
@@ -199,7 +200,8 @@ public void testStartLocalizer() throws IOException {
       Assert.assertEquals(result.get(13),
           "-Dlog4j.configuration=container-log4j.properties" );
       Assert.assertEquals(result.get(14),
-          "-Dyarn.app.container.log.dir=${yarn.log.dir}/userlogs/application_0/12345");
+          String.format("-Dyarn.app.container.log.dir=%s/application_0/12345",
+          mockExec.getConf().get(YarnConfiguration.NM_LOG_DIRS)));
       Assert.assertEquals(result.get(15),
           "-Dyarn.app.container.log.filesize=0");
       Assert.assertEquals(result.get(16), "-Dhadoop.root.logger=INFO,CLA");
@@ -278,7 +280,7 @@ public Object answer(InvocationOnMock invocationOnMock)
           }
         }
     ).when(container).handle(any(ContainerDiagnosticsUpdateEvent.class));
-    
+
     when(cId.toString()).thenReturn(containerId);
 
     when(context.getEnvironment()).thenReturn(env);
@@ -312,7 +314,7 @@ public void testInit() throws Exception {
     
   }
 
-  
+
   @Test
   public void testContainerKill() throws IOException {
     String appSubmitter = "nobody";
@@ -320,7 +322,7 @@ public void testContainerKill() throws IOException {
         LinuxContainerExecutor.Commands.SIGNAL_CONTAINER.getValue());
     ContainerExecutor.Signal signal = ContainerExecutor.Signal.QUIT;
     String sigVal = String.valueOf(signal.getValue());
-    
+
     mockExec.signalContainer(appSubmitter, "1000", signal);
     assertEquals(Arrays.asList(YarnConfiguration.DEFAULT_NM_NONSECURE_MODE_LOCAL_USER,
         appSubmitter, cmd, "1000", sigVal),
-- 
1.7.9.5

